src/overlaybd/tar/tarerofs.cpp

```diff
@@ -0,0 +1,113 @@
+#include "tarerofs.h"
+#include <stdio.h>
+#include <string.h>
+#include <sys/param.h>
+#include <sys/types.h>
+#include <sys/sysmacros.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <cstdio>
+#include <utime.h>
+#include <set>
+#include <string>
+#include <photon/fs/localfs.h>
+#include <photon/fs/path.h>
+#include <photon/common/string_view.h>
+#include <photon/fs/filesystem.h>
+#include <photon/common/alog.h>
+#include <photon/fs/fiemap.h>
+#include "../lsmt/file.h"
+#include "../lsmt/index.h"
+
+int TarErofs::extract_all() {
+    ssize_t read;
+    struct stat st;
+    char buf[128*1024];
+    char base_path[64] = "/tmp/tarerofs_base_XXXXXX";
+    char command_line[256] = "mkfs.erofs --tar=0,upper.map,1073741824 --aufs";
+    char command_line2[] = " upper.erofs";
+
+    FILE *fp;
+    photon::fs::IFile *rawfs = nullptr;
+    int status;
+    uint64_t blkaddr, toff;
+    uint32_t nblocks;
+
+    if (!meta_only) {
+         LOG_ERROR("currently erofs supports fastoci mode only", strerror(errno));
+         return -1;
+    }
+
+    if (fs_base_file->fstat(&st)>= 0 && st.st_blocks > 0) {
+        int fd = mkstemp(base_path);
+        if (fd < 0) {
+            LOG_ERROR("cannot generate a temporary file to dump overlaybd disk");
+            return -1;
+        }
+        std::strcat(command_line, " --base ");
+        std::strcat(command_line, base_path);
+
+        for (int i = 0; i < 800; i++) {
```

Comment:
800 here is quite hard-coded. But I'm not sure how we could detect the real written data in the first segment(extent) since we don't need the following `LSMT::IFileRW::RemoteData` ranges.